spring-cloud-gateway-server/src/main/java/org/springframework/cloud/gateway/config/GatewayAutoConfiguration.java

```diff
@@ -292,6 +298,24 @@ public GRPCResponseHeadersFilter gRPCResponseHeadersFilter() {
 		return new GRPCResponseHeadersFilter();
 	}
 
+	@Bean
+	@ConditionalOnEnabledFilter
+	@ConditionalOnProperty(name = "server.http2.enabled", matchIfMissing = true)
+	public JsonToGrpcGatewayFilterFactory jsonToGRPCFilterFactory(GRPCSSLContext gRPCSSLContext) {
+		return new JsonToGrpcGatewayFilterFactory(gRPCSSLContext);
+	}
+
+	@Bean
+	@ConditionalOnEnabledFilter(JsonToGrpcGatewayFilterFactory.class)
+	@ConditionalOnMissingBean(GRPCSSLContext.class)
+	public GRPCSSLContext gRPCSSLContext() throws KeyStoreException, NoSuchAlgorithmException {
```

Comment:
This should also have a conditional on class for some class in grpc, `Channel` or `ClientCall`